### ISSUE TITLE
Github Updater for Pressbooks 5.35 Release Fails

### ISSUE THREAD

**Issue by beckej13820** [pressbooks/pressbooks#2818]
This issue is specifically happening trying to update an Open Source version of Pressbooks with Github Updater.
![Screen Shot 2022-06-06 at 4 23 02 PM](https://user-images.githubusercontent.com/29982981/172242838-0aa241a2-5b2f-4229-b53e-9adebfa34fbd.png)

I do not think the package is properly encoded for the Github updater to use and it is showing the error "Update package not available."

Once upon a time, Github updater was the recommended plugin for keeping your Pressbooks installation up to date. Please let me know if it is no longer supported and I will stop using it.

### Steps to Reproduce

1. Try to update from Pressbooks 5.34.x to Pressbooks 5.35 using the github updater.
2. Error message

**Expected behavior:** Typically, this works very well.

**Actual behavior:** No upgrade has occurred.

#### Root Blog Info

Root Blog ID: 1
Root Blog URL: https://pressbooks.sunycreate.cloud/

#### Browser

Platform: OS X
Browser Name: Chrome
Browser Version: 102.0.0.0
User Agent String: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36

#### WordPress Configuration

Network URL: https://pressbooks.sunycreate.cloud/
Network Type: Subdomain
Version: 6.0
Language: en_US
WP_ENV: Not set
WP_DEBUG: Enabled
Memory Limit: 64M

#### Pressbooks Configuration

Version: 5.34.1
Root Theme: Aldine
Root Theme Version: 1.14.1

#### Pressbooks Dependencies

Epubcheck: Installed
xmllint: Installed
PrinceXML: Installed
Saxon-HE: Installed

#### Must-Use Plugins

hm-autoloader.php: n/a

#### Network Active Plugins

Git Updater: 11.0.3
Limit Login Attempts Reloaded: 2.25.3
Pressbooks: 5.34.1
Pressbooks Stats: 1.6.4

#### Root Blog Active Plugins

Git Updater: 11.0.3
Limit Login Attempts Reloaded: 2.25.3
Pressbooks: 5.34.1
Pressbooks Stats: 1.6.4

#### Inactive Plugins

Candela Citation: 0.2.4
H5P: 1.15.4
Hypothesis: 0.6.0
Pressbooks Shortcode Handler: 0.1
TablePress: 1.14
WordPress Importer: 0.7
WP QuickLaTeX: 3.8.6

#### Server Configuration

PHP Version: 8.0.7
MySQL Version: 5.7.30
Webserver Info: Apache

#### PHP Configuration

Memory Limit: 1024M
Upload Max Size: 128M
Post Max Size: 256M
Upload Max Filesize: 128M
Time Limit: 30
Max Input Vars: 1000
URL-aware fopen: On (1)
Display Errors: N/A

#### PHP Extensions

OPcache: Disabled
XDebug: Disabled
cURL: Supported
cURL Version: 7.71.0
imagick: Installed
xsl: Installed

**Comment by beckej13820** [comment]
More information about this bug.

Upgrade also fails if you download the zip file from github and upload it to your server via FTP.

If you do that, the error message when you visit pressbooks is that dependencies are missing. I was able to go to my command line interface and run `composer install`, which fixed the problem.

I think something is going wrong with your creation process with the zip files.

**Comment by SteelWagstaff** [comment]
@beckej13820 we just released 5.35.1 a few minutes ago. Does it have same problem for you?

**Comment by beckej13820** [comment]
5.35.1 was no problem. It only took a little while because I backed up 4 times before hitting the button. ;-)

**Comment by SteelWagstaff** [comment]
I don't think we did anything different, but I'm not 100% sure. Ho Man usually handles the releases but was out today so I did the release & packaging myself. I'll ask him when he's back at work ...

**Comment by SteelWagstaff** [comment]
Ed -- we didn't do anything different that we can think of with these two releases. Let's keep an eye on this for the next couple releases and see if it reoccurs and if so, why?

**Comment by beckej13820** [comment]
Just popping in here to this thread to say that the most recent Pressbooks release, 5.36.1, fails with the github updater.

The error message says "An error occurred while updating Pressbooks: Update package not available."
![Screen Shot 2022-07-15 at 9 42 15 AM](https://user-images.githubusercontent.com/29982981/179235170-bedd016b-1599-4a7d-afb3-f5b7b0025939.png)

Leaving the Pressbooks interface, I went directly to github to inspect the .zip file that the github updater pulls and uses. I just did a short inspection to try to see if I would notice anything different between the code in the Github repository and the .zip file. I was looking for files or folders that were missing, because of my gut feeling that updates were failing when dependencies weren't included into the package.

Comparing https://github.com/pressbooks/pressbooks to https://github.com/pressbooks/pressbooks/archive/refs/tags/5.36.1.zip

- Pressbooks/assets/src is not included in the zip file
- Pressbooks/bin is not included in the zip file

I'm not sure if it is related, but I thought I would share.

-

**Comment by SteelWagstaff** [comment]
Thanks @beckej13820 this one is one us. See #2883. We need to fix our tests and then run the build/deploy step in our CI auto-pipeline to fix this for GitHub Updater users.

**Comment by beckej13820** [comment]
Also fails for 5.37.0
